### PR TITLE
feat: make quackstack package-ready

### PR DESCRIPTION
**Pull request by vcokltfre**
This PR makes quackstack ready to be used as an importable package and ready to be distributed on PyPI. This is a draft PR for the time being because I intend to also add the command-line generation as part of this PR, in addition to writing substantially more documentation on how to use both the HTTP API and programmatic API.

### CODE REVIEW

**Comment by vcokltfre** [comment]
After consideration, making a CLI should be done as part of a separate PR, so this is now marked as ready for review.

**Comment by Shivansh-007** [comment]
I think @Akarys42 was working on the CLI for quackstack, could you update us on the progress?

**Comment by Akarys42** [comment]
> I think @Akarys42 was working on the CLI for quackstack, could you update us on the progress?

I honestly totally forgot about it, I'll unassign me from the issues to stop blocking further progress.

**Review by Shivansh-007 — APPROVED** [review]
- Could you add `Response` to the `get_details` function return type annotation, it is returned when the `type` is an invalid one.

Otherwise, the change looks good :shipit:

File: quackstack/manducky.py [hunk 1]
```diff
@@ -101,18 +100,18 @@ def generate_from_options(self, options: dict) -> dict:
...
-            variation_=options['variation']
+            variation_=options["variation"]
         )
 
     def generate(
         self,
         *,
-            options: Optional[ManDuckRequest] = None,
+            options: Optional[dict] = None,
             ducky: Optional[ProceduralDucky] = None
```
Comment (Shivansh-007) [inline quackstack/manducky.py:110]: Missed this out earlier, could you remove the extra-indentation here.

**Comment by jb3** [comment]
We're not going to proceed with packaging Quackstack at this point. Integrations such as Sir Lancebot and potentially a CLI in future will call out to the publicly hosted instance at quackstack.pythondiscord.com.

**Comment by jb3** [comment]
Reopened because there are still some amendments here which may be nice to have even if not distributing to PyPI.

**Review by Bluenix2 — APPROVED** [review]
Looks good

**Pull request merged**